pkg/scheduler/core/generic_scheduler.go

```diff
@@ -653,28 +653,19 @@ func PrioritizeNodes(
 
 	results := make([]schedulerapi.HostPriorityList, len(priorityConfigs), len(priorityConfigs))
 
-	for i, priorityConfig := range priorityConfigs {
-		if priorityConfig.Function != nil {
-			// DEPRECATED
-			wg.Add(1)
-			go func(index int, config algorithm.PriorityConfig) {
-				defer wg.Done()
-				var err error
-				results[index], err = config.Function(pod, nodeNameToInfo, nodes)
-				if err != nil {
-					appendError(err)
-				}
-			}(i, priorityConfig)
-		} else {
-			results[i] = make(schedulerapi.HostPriorityList, len(nodes))
-		}
+	for i := range priorityConfigs {
+		results[i] = make(schedulerapi.HostPriorityList, len(nodes))
 	}
 
 	processNode := func(index int) {
 		nodeInfo := nodeNameToInfo[nodes[index].Name]
 		var err error
-		for i := range priorityConfigs {
+		for i, priorityConfig := range priorityConfigs {
```

Comment:
Thanks, @zhangmingld for your PR, but this change is not equivalent to the original code. In the original code, each old-style priority function is called once for all the nodes (line 663 of the original code).
The `processNode` function at line 673 of the original code is called once per node. By moving the logic inside `processNode`, you would make the priority functions be unnecessarily called once per node.

Reply:
I think i had done this, the line "if results[i][0].Host == """, this will make sure that the priority runs once per node . For example , we have node1 and node3, in the beginning,results[i] is empty ,after node1 completes, then the results[i] will have some value , now node3 starts , it will found  results[i] is not empty ,so it will not calculate again. Though node1 and node3 will run in Parallelize, but in fact ,they won`t start in the same time, so i think it will work

Reply:
I see what you have done now. At the very least, you should add a comment explaining this.